File: app/src/main/java/com/example/lucas/haushaltsmanager/Database/Repositories/Bookings/BookingDAO.java
package com.example.lucas.haushaltsmanager.Database.Repositories.Bookings;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;
import androidx.room.Update;

import com.example.lucas.haushaltsmanager.entities.Booking.BookingWithCategory;
import com.example.lucas.haushaltsmanager.entities.Booking.BookingWithoutCategory;

import java.util.List;
import java.util.UUID;

@Dao
public abstract class BookingDAO {
    @Query("SELECT * FROM bookings WHERE id = :id")
    public abstract BookingWithoutCategory get(UUID id);

    @Transaction
    @Query("SELECT * FROM bookings WHERE id = :id")
    public abstract BookingWithCategory getBookingWithCategory(UUID id);

    @Transaction
    @Query("SELECT * FROM bookings")
    public abstract List<BookingWithCategory> getBookingsWithCategory();

    @Query("SELECT * FROM bookings")
    public abstract List<BookingWithoutCategory> getAll();

    @Insert
    public abstract void insert(BookingWithoutCategory expense);

    @Delete
    public abstract void delete(BookingWithoutCategory expense);

    @Update
    public abstract void update(BookingWithoutCategory expense);

    public void insert(BookingWithCategory booking) {
        insert(booking.getBooking());
    }

    public void update(BookingWithCategory booking) {
        update(booking.getBooking());
    }
//
//    void insert(ParentBooking booking);
}

File: app/src/main/java/com/example/lucas/haushaltsmanager/Database/Repositories/BookingDAO.kt
package com.example.lucas.haushaltsmanager.Database.Repositories

import androidx.room.*
import androidx.sqlite.db.SupportSQLiteQuery
import com.example.lucas.haushaltsmanager.entities.booking.Booking
import java.util.*

@Dao
interface BookingDAO {
    @Query("SELECT * FROM bookings")
    fun getAll(): List<Booking>

    @RawQuery
    fun getFilteredList(query: SupportSQLiteQuery): List<Booking>

    @Query("SELECT * FROM bookings WHERE account_id IN (:accountIds)")
    fun getAllWithAccounts(accountIds: List<UUID>): List<Booking>

    @Insert
    fun insert(booking: Booking)

    @Update
    fun update(booking: Booking)

    @Delete
    fun delete(booking: Booking)
}